lib/sift/resource.rb

```diff
@@ -0,0 +1,61 @@
+
+module Sift
+
+  module Resource
+
+    def self.included(klass)
+      klass.extend(ClassMethods)
+      klass.include(InstanceMethods)
+    end
+
+    module ClassMethods
+
+      BASE_URI = "https://api3.siftscience.com/v3/accounts/"
+
+      def resource_uri(path)
+        BASE_URI + Sift.account_id + path
+      end
+
+      def post(uri, options = {})
+      end
+
+      def put(uri, options = {})
+        http_request do 
+          HTTParty.put(uri, options.merge(default_options))
+        end
+      end
+
+      def get(uri, options = {})
+        http_request do 
+          HTTParty.get(uri, options.merge(default_options))
+        end
+      end
+
+    private
+      def default_options
+        {
+          headers: { 
+            "Authorization" => "Basic #{Sift.api_key}", 
+            "User-Agent" => "sift-ruby/#{Sift::VERSION}", 
+            "Content-Type" => "application/json", 
+          },
+        }
+      end
+
+      def http_request(&blk)
+        response = blk.call
+        if response.success?
+          response.body
+        else
+          raise(RuntimeError, "Request failed: #{response.code}\n#{response.body}", caller)
```

Comment:
This isn't very friendly — raising a generic RuntimeException limits clients' ability to handle errors gracefully. There's also no easy way for them to get back the status code.

We may want to define our own exception hierarchy.